evalkit_api_client/evalkit_api_client/v1_client.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import requests
+from requests import Response
+from typing import Any, Dict, Iterator, Optional
+from urllib.parse import urljoin
+
+from evalkit_api_client.errors import APIPaginationException
+from evalkit_api_client.interface import EvalKitAPIClient
+from evalkit_api_client.types import RequestHeaders, RequestParams
+
+logger = logging.getLogger()
+
+
+class EvalKitAPIv1(EvalKitAPIClient):
+    """
+    Evalkit v1 API Client.
+
+    This client can be used to make requests to the v1 Evalkit API.
+
+    Create separate clients for other versions of the Evalkit API.
+    """
+
+    def __init__(self,
+                 api_url: str,
+                 api_token: Optional[str] = None) -> None:
+        """
+        Creates a evalkit API client given a base URL for the API, an optional
+        API token, and an optional requests library.
+
+        The optional requests library should be either the python HTTP requests
+        library or the equivalent (e.g. a Requests-OAuthlib session object).
+        """
+        self._api_url = api_url
+        self._api_token = api_token
+
+    def _get_url(self, endpoint: str) -> str:
+        """
+        Formats a EvalKit API URL from a given endpoint.
+        """
+        return urljoin(self._api_url, endpoint)
+
+    def _add_bearer_token(self, headers: Dict[str, Any]):
+        """
+        Adds the authentication bearer token. Only run this if the token
+        exists.
+        """
+        headers.update({'AuthToken': self._api_token})
+
+    def _send_request(self,
+                      callback,
+                      url: str,
+                      exit_on_error: bool = True,
+                      headers: RequestHeaders = None,
+                      params: RequestParams = None,
+                      **kwargs) -> Response:
+        """
+        Sends an API call to the Evalkit server via callback method.
+
+        The callback should be a requests.<func> function or the equivalent.
+
+        Note: since raise_for_status() will raise an exception for error
+        codes, the user is responsible for catching `HTTPError`
+        exceptions unless they run with the exit_on_error set to False.
+        """
+        if headers is None:
```

Comment:
Is this preferable to `if not headers`?

Reply:
yes, i think so because it only evaluates as truthy in one case

Reply:
sure we could catch situations where its `0` or `1` and `False` or `True` but that would lead to a `headers` argument that can be set in a lot of different ways, which could become confusing.

Reply:
It will also fail in weird cases the way I have it now, meaning if someone passes a value of `1` the requests library will bomb out. But at least it only defaults to empty headers when it is `None` (which is also the default). This seems to be not too surprising to me. 

Whereas if I pass `headers=0` and it works, that's somewhat surprising.

Reply:
No, go for it!

Reply:
leaving as is!